go/vt/key/key.go

```diff
@@ -49,261 +52,258 @@ func (i Uint64Key) Bytes() []byte {
 	return buf
 }
 
-//
-// KeyRange helper methods
-//
+// Helper methods for keyspace id values.
 
-// EvenShardsKeyRange returns a key range definition for a shard at index "i",
-// assuming range based sharding with "n" equal-width shards in total.
-// i starts at 0.
-//
-// Example: (1, 2) returns the second out of two shards in total i.e. "80-".
-//
-// This function must not be used in the Vitess code base because Vitess also
-// supports shards with different widths. In that case, the output of this
-// function would be wrong.
-//
-// Note: start and end values have trailing zero bytes omitted.
-// For example, "80-" has only the first byte (0x80) set.
-// We do this to produce the same KeyRange objects as ParseKeyRangeParts() does.
-// Because it's using the Go hex methods, it's omitting trailing zero bytes as
-// well.
-func EvenShardsKeyRange(i, n int) (*topodatapb.KeyRange, error) {
-	if n <= 0 {
-		return nil, fmt.Errorf("the shard count must be > 0: %v", n)
-	}
-	if i >= n {
-		return nil, fmt.Errorf("the index of the shard must be less than the total number of shards: %v < %v", i, n)
-	}
-	if n&(n-1) != 0 {
-		return nil, fmt.Errorf("the shard count must be a power of two: %v", n)
+// Normalize adds padding to id as necessary to make sure it can be compared as an 8 byte integer.
+// From a key range perspective, -80 == 00-80 == 0000-8000 == 000000-800000, etc. and they should
+// always be treated the same even if they are different lengths.
+func Normalize(id []byte) []byte {
+	if len(id) >= 8 {
+		return id[:8]
```

Comment:
I'm not sure that this is expected or correct behavior. 

Wouldn't it be better to do this?
```
func Normalize(id []byte) ([]byte, error) {
            if !IsValidKey(id) {
                return fmt.Errorf("provided keyspace ID %q is outside of the valid unsigned 64 bit range", hex.EncodeToString(id))
```

We'd then have to thread that back through compare. This ties into my error handling comment in `TestCompare`.

What do you think?